cmake/OpenCVDownload.cmake

```diff
@@ -0,0 +1,124 @@
+#
+#  Download and optionally unpack a file
+#
+#  ocv_download(FILENAME p HASH h URL u DESTINATION_DIR d [STATUS s] [UNPACK] [RELATIVE])
+#    FILENAME - filename
+#    HASH - MD5 hash
+#    URL - full download url
+#    DESTINATION_DIR - file will be copied to this directory
+#    STATUS - passed variable will be updated in parent scope,
+#             function will not fail the build in case of download problem if this option is provided,
+#             but will fail in case when other operations (copy, remove, etc.) failed
+#    UNPACK - downloaded file will be unpacked to DESTINATION_DIR
+#    RELATIVE - if set, then URL is treated as base, and FILENAME will be appended to it
+#  Note: uses OPENCV_DOWNLOAD_DIR folder as cache, default is <opencv>/.cache
+
+#
+# Main function
+#
+
+set(OPENCV_DOWNLOAD_DIR "${OpenCV_SOURCE_DIR}/.cache" CACHE PATH "Download cache dir")
```

Comment:
We need to support environment variables to specify external download directory - It is preferable way for buildbot